R/derive_vars_extreme_date.R

```diff
@@ -0,0 +1,154 @@
+#'
+#'   Only observations of the dataset which fulfill the specified condition
+#'   are considered for the treatment start date.
+#'
+#'   Default: NULL
+#'
+#'   Permitted Values: logical expression
+#'
+#' A list of quosures where the expressions are symbols as returned by
+#' `vars()` is expected.
+#'
+#' @param dtc name of variable from which the extreme date is derived e.g. `EXENDTC`
+#'
+#' @param new_var name of the extreme date derived e.g.treatment end date `TRTEDTM`
+#'
+#' @inheritParams impute_dtc
+#'
+#' @details For each group (with respect to the variables specified for the
+#'   `by_vars` parameter) the first observation (with respect to the order
+#'   specified for the `order` parameter) is included in the output dataset.
+#'
+#' @author Stefan Bundfuss, Teckla Akinyi
+#'
+#' @return The input dataset with extreme date variable added e.g. `TRTEDTM`
+#'
+#' @export
+#'
+#' @keywords adsl timing derivation
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' data("ex")
+#' data("dm")
+#'
+#'dm %>%
+#'  derive_vars_extreme_date(subject_keys=vars(STUDYID,USUBJID),
+#'                           dataset_date = ex,
+#'                           filter_dataset_date = EXDOSE>0,
+#'                           dtc = EXSTDTC,
+#'                           order = vars(EXSTDTC,EXSEQ),
+#'                           new_var=TRTEDTM,
+#'                           date_imputation = "first",
+#'                           time_imputation = "first",
+#'                           flag_imputation = "AUTO",
+#'                           min_dates = NULL,
+#'                           max_dates = NULL,
+#'                           mode="LAST") %>%
+#'   select(USUBJID, TRTEDTM)
+derive_vars_extreme_date <- function(dataset,
+                                     subject_keys =vars(STUDYID,USUBJID),
+                                     dataset_date,
+                                     filter_dataset_date = NULL,
+                                     dtc,
+                                     order,
+                                     new_var,
+                                     date_imputation,
+                                     time_imputation,
+                                     flag_imputation = "AUTO",
+                                     min_dates = NULL,
+                                     max_dates = NULL,
+                                     mode) {
+
+  new_var <- assert_symbol(enquo(new_var))
+  dtc <- assert_symbol(enquo(dtc))
+  assert_data_frame(dataset, subject_keys)
+  assert_data_frame(dataset_date, required_vars = quo_c(subject_keys, order))
+  assert_character_scalar(mode, values = c("first", "last"), case_sensitive = F)
+
+  filter_dataset_date <- assert_filter_cond(enquo(filter_dataset_date), optional = TRUE)
+  filter_dataset_date <- enquo(filter_dataset_date)
+
+  assert_character_scalar(flag_imputation, values = c("auto", "both", "date", "time", "none"),
```

Comment:
To make `flag_imputation` case insensitive `flag_imputation <- assert_character_scalar(...` should be used. (`assert_character_scalar()` returns the lower case value if `case_sensitive = FALSE` is specified.)